### ISSUE TITLE
Repack seems to only works on valid delimiters?

### ISSUE THREAD

**Issue by Kn0rk** [cursorless-dev/cursorless#1751]
![image](https://github.com/cursorless-dev/cursorless/assets/12850488/281a891a-21dc-4627-8885-b15b7656c902)

`quad repack [W]hiskey `

Will replace the curly brackets around the function rather than the single quotes around the (fake) string. In Java single quotes are only used for characters and double quotes are used for strings. As such defining a string with single quotes is incorrect. I am assuming that the delimiter search is based on the parser which does not recognize the single quotes as delimiters.

### To reproduce:

Create a java file in the vscode and paste:
```
public class TestSingleQuoteDelimiter {
    
    public static void main(String[] args) {
        String notAString = 'this will not work';
    }
  
}
```
The result after the command is:

```
public class TestSingleQuoteDelimiter {
    
    public static void main(String[] args) "
        String notAString = 'this will not work';
    "
    
}

```

**Comment by pokey** [comment]
@AndreasArvidsson would your https://github.com/cursorless-dev/cursorless/issues/1600 help here?

**Comment by Kn0rk** [comment]
Yes, I think that should fix it. Since tree sitter would not know what to do with the single quotes they should remain as valid delimiters. 

Srry I didn’t find that issue when i was searching for similar issues.

**Comment by AndreasArvidsson** [comment]
I think so

**Comment by AndreasArvidsson** [comment]
Now supported
